finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/Message.scala

```diff
@@ -140,6 +140,9 @@ object Response {
   def apply(headers: Headers, stream: Stream): Response =
     Impl(headers, stream)
 
+  def unapply(arg: Response): Option[Status] =
```

Comment:
Since `arg.status` throws an exception instead of returning `null`  based on the `arg.status` definition
```scala
override def status = headers.get(Headers.Status).headOption match {
      case Some(code) => Status.fromCode(code.toInt)
      case None => throw new IllegalArgumentException(s"missing ${Headers.Status} header")
    }
```

Would it work to do this instead?
```scala
def unapply(arg: Response): Option[Status] =
    Some(arg.status)
```

Reply:
`Some(arg.status)` works.